maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java

```diff
@@ -523,15 +535,122 @@ private File loadProjectProperties(Builder builder, MavenProject bndProject, Mav
 		if (baseDir != null) {
 			builder.updateModified(pomFile.lastModified(), "POM: " + pomFile);
 		}
+
+		boolean bndUsed = false;
 		Xpp3Dom bndElement = configuration.getChild("bnd");
 		if (bndElement != null) {
-			logger.debug("loading bnd properties from bnd element in pom: {}", pomProject);
-			UTF8Properties properties = new UTF8Properties();
-			properties.load(bndElement.getValue(), pomFile, builder);
-			// we use setProperties to handle -include
-			builder.setProperties(baseDir, properties.replaceHere(baseDir));
+			if (projectFile == null) {
+				if (!xppDomEqualsWithSubtitution(bndElement, tracker.get("bnd"), projectFile, builder)) {
+					tracker.put("bnd", bndElement);
+					bndUsed = true;
+					logger.debug("loading bnd properties from bnd element in pom: {}", pomProject);
+					projectFile = pomFile;
+					UTF8Properties properties = new UTF8Properties();
+					properties.load(bndElement.getValue(), projectFile, builder);
+					// we use setProperties to handle -include
+					builder.setProperties(baseDir, transformUnderscoreKeys(properties.replaceHere(baseDir)));
+				}
+			} else {
+				logger.warn("Pom defines both bndfile and bnd configuration. Ignoring the bnd configuration in pom: {}",
+					pomProject);
+			}
+		}
+
+		if (instructions != null && instructions.size() > 0 && configuration.getChild("instructions") != null) {
+			if (projectFile == null && !bndUsed) {
+				logger.debug("loading bnd properties from the instructions element in pom: {}", pomProject);
+				projectFile = pomFile;
+				// Copy into UTF8Properties so that we can replace ${.}
+				// placeholders
+				UTF8Properties properties = new UTF8Properties();
+
+				properties.putAll(instructions);
+				builder.setProperties(baseDir, transformUnderscoreKeys(properties.replaceHere(baseDir)));
+			} else {
+				logger.warn(
+					"Pom defines both a bnd/bndfile and instructions element. Ignoring the instructions element in pom: {}",
+					pomProject);
+			}
+		}
+
+		return projectFile;
+	}
+
+	/**
+	 * Compare two Xpp3Dom objects, substituting any variable placeholders if
+	 * available.
+	 * 
+	 * @param d1 The first XML object.
+	 * @param d2 The seconds XML object.
+	 * @param projectFile A properties file used by the properties loader for
+	 *            error reporting.
+	 * @param reporter A reporter for logging.
+	 * @return {@code true} if the two XML objects are equal after placeholder
+	 *         substitution.
+	 * @throws IOException If an IO exception occurs.
+	 */
+	private boolean xppDomEqualsWithSubtitution(Xpp3Dom d1, Xpp3Dom d2, File projectFile, Reporter reporter)
```

Comment:
xppDomEqualsWithSubtitution -> xppDomEqualsWithSubstitution

Reply:
Thanks for spotting! I will update this.